src/plugins/paste-image/index.ts

```diff
@@ -0,0 +1,41 @@
+import { PluginController } from "../PluginController";
+
+export default class PasteImage extends PluginController {
+  static id = "paste-image" as const;
+  static enabledByDefault = true;
+
+  pasteHandler = this._pasteHandler.bind(this);
+
+  afterEnable() {
+    document.addEventListener("paste", this.pasteHandler);
+  }
+
+  afterDisable() {
+    document.removeEventListener("paste", this.pasteHandler);
+  }
+
+  _pasteHandler(e: ClipboardEvent) {
+    const clipboardFiles = e.clipboardData?.files;
+    if (clipboardFiles?.length) {
+      e.preventDefault();
+      if (!this.calc.selectedExpressionId) {
+        // avoid images being inserted at the top of the expressions list
+        // when there is no selected expression
+        if (document.activeElement !== document.body) return;
+        const lastExpression = this.calc.getExpressions().splice(-1)[0];
+        lastExpression &&
```

Comment:
Better to make the control flow more obvious.
```suggestion
        if (lastExpression)
```
 `&&` would be good on an assignment RHS like `const id = lastExpression && lastExpression.id` (though that's typically equivalent to `const id = lastExpression?.id`. Would only need a `&&` if it's a construction like `const id = lastExpression && normalizeID(lastExpression.id)`).